src/main.ts

```diff
@@ -694,6 +694,25 @@ export class Purchases {
     return this._appUserId;
   }
 
+  /**
+   * Sets attributes for the current user. Attributes are useful for storing additional, structured information on a customer that can be used elsewhere in the system.
+   * For example, you could store your customer's email address or additional system identifiers through the applicable reserved attributes, or store arbitrary facts like onboarding survey responses, feature usage, or other dimensions as custom attributes.
+   * @param attributes - A dictionary of attributes to set for the current user.
+   * @throws {@link PurchasesError} if there is an error while setting the attributes or if the customer doesn't exist.
+   */
+  public async setAttributes(attributes: {
+    [key: string]: string | null;
+  }): Promise<void> {
+    /*
+     * Ensure the customer exists by calling getCustomerInfo as setAttributes will throw an error
+     * if the customer doesn't exist.
+     *
+     * Note: We may want to optimize this in the future by prefetching customer info during SDK initialization.
+     */
```

Comment:
or at least we could store a boolean in the instance if we've called `getCustomerInfo()` once, so subsequent attr calls don't have to call it again. 

NBD anyway as it will be a fast call

Reply:
I'm going to leave this optimization for later